src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -5261,14 +5340,169 @@ void wm_vuldet_free_agent_software(agent_software *agent) {
     }
 }
 
-int wm_vuldet_fetch_MSU() {
-    if (w_uncompress_gzfile(VU_MSU_FILE_GZ, VU_FIT_TEMP_FILE)) {
-        return VU_INV_FEED;
+int wm_vuldet_fetch_MSU(update_node *update, char *repo) {
+    char *hash = NULL;
+
+    if (*repo == '\0') {
+        repo = MSU_REPO;
+        if (wm_vuldet_update_MSU(update) == false) {
+            return VU_NOT_NEED_UPDATE;
+        }
+        hash = wm_vuldet_get_hash(vu_feed_tag[update->dist_tag_ref]);
+        if (hash == NULL) {
+            merror("Failed to retrieve hash value. File integrity won't be checked.");
+        }
     }
+ 
+    int res_url_request;
+
+    for (int8_t attempts = 0;; attempts++) {
+        res_url_request = wurl_request_gz(repo, VU_FIT_TEMP_FILE, NULL, NULL, update->timeout, hash);
```

Comment:
Can we use `wurl_request_uncompress_bz2_gz()` instead? they are equivalent and it adds compatibility to bz2 format